chain.go

```diff
@@ -352,6 +353,84 @@ func (c *Client) GetBlockHeaderVerbose(hash *chainhash.Hash) (*dcrjson.GetBlockH
 	return c.GetBlockHeaderVerboseAsync(hash).Receive()
 }
 
+// FutureGetBlockSubsidyResult is a future promise to deliver the result of a
+// GetBlockSubsidyAsync RPC invocation (or an applicable error).
+type FutureGetBlockSubsidyResult chan *response
+
+// Receive waits for the response promised by the future and returns a data
+// structure of the block subsidy requested from the server given its height
+// and number of voters.
+func (r FutureGetBlockSubsidyResult) Receive() (*dcrjson.GetBlockSubsidyResult, error) {
+	res, err := receiveFuture(r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal the result
+	var bs dcrjson.GetBlockSubsidyResult
+	err = json.Unmarshal(res, &bs)
+	if err != nil {
+		return nil, err
+	}
+	return &bs, nil
+}
+
+// GetBlockSubsidyAsync returns an instance of a type that can be used to
+// get the result of the RPC at some future time by invoking the Receive
+// function on the returned instance.
+//
+// See GetBlockSubsidy for the blocking version and more details.
+func (c *Client) GetBlockSubsidyAsync(height int64, voters uint16) FutureGetBlockSubsidyResult {
+	cmd := dcrjson.NewGetBlockSubsidyCmd(height, voters)
+	return c.sendCmd(cmd)
+}
+
+// GetBlockSubsidy returns a data structure of the block subsidy
+// from the server given its height and number of voters.
+func (c *Client) GetBlockSubsidy(height int64, voters uint16) (*dcrjson.GetBlockSubsidyResult, error) {
+	return c.GetBlockSubsidyAsync(height, voters).Receive()
+}
+
+// FutureGetCoinSupplyResult is a future promise to deliver the result of a
+// GetCoinSupplyAsync RPC invocation (or an applicable error).
+type FutureGetCoinSupplyResult chan *response
+
+// Receive waits for the response promised by the future and returns the
+// current coin supply
+func (r FutureGetCoinSupplyResult) Receive() (*dcrutil.Amount, error) {
+	res, err := receiveFuture(r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal the result
+	var cs float64
+	err = json.Unmarshal(res, &cs)
+	if err != nil {
+		return nil, err
+	}
+	amt, err := dcrutil.NewAmount(cs)
```

Comment:
Since `getcoinsupply` returns atoms, as mentioned elsewhere, this should just be a simple cast from an int64.

```Go
var cs int64
err = json.Unmarshal....
...
return dcrutil.Amount(cs), nil
```

Also, please actually test these before submitting changes.  If you had tested it you would've noticed that the value was incorrect.